pgcli/pgexecute.py

```diff
@@ -233,7 +233,7 @@ def connect(self, database=None, user=None, password=None, host=None,
         }
         new_params.update(kwargs)
         conn_params.update({
-            k: unicode2utf8(v) for k, v in new_params.items() if v is not None
+            k: unicode2utf8(v) for k, v in new_params.items() if v
```

Comment:
Would this change alone be enough to fix the problem, or `if not dsn:` change is necessary as well?

Reply:
Both changes are necessary. If the dsn check removed, both user and database will be present on conn_params, triggering the issue.